Paradigm4/SciDB-Py
# Upgrade to Pandas 2.1.0

### Code Fixes

* Pandas up to `2.1.0` set in `requirements.txt` and `setup.py`
* Use new `datetime64[ns, UTC]` Pandas type for SciDB `datetimetz` type in schema promotion to Pandas

### Test Fixes

* Skip `dataframe_promo=False` example due to bub in Pandas reported [here](https://github.com/pandas-dev/pandas/issues/55011)
* Skip SSL examples for CentOS 7 and Python 3.10 setups as the Shim certificate generated on CentOS 7 is not accepted in Python 3.10 see [here](https://bugs.python.org/issue43998) and [here](https://github.com/python/cpython/pull/25778) 
* Replace `numpy.array` with `numpy.rec.array` so they can be compared
* Use `equals` and `is` to compare arrays instead of `==`
* Remove `typed_array_obj` as is not used anymore
